### ISSUE TITLE
[fluorine] arch beacon failures

### ISSUE THREAD

**Issue by Ch3LL** [saltstack/salt-jenkins#1224]
integration.states.test_beacon.BeaconStateTestCase.test_present_absent

https://jenkinsci.saltstack.com/job/fluorine/view/Python3/job/salt-arch-py3/48/

**Comment by terminalmage** [comment]
Haven't been able to reproduce this in Docker or in AWS (via testkitchen) when running a single test module at a time. I'm presently running the entire suite in TK.

**Comment by terminalmage** [comment]
So, the test is failing because inside the `beacons` execution module the function `list_available` is run, which tries to list the available beacons, and the list is empty.

The reason it is empty is because the way that it gets the list of available beacons is by firing an event on the minion bus (using `event.fire`) and reading the response event. However, the act of firing the event itself fails with the error `Failed to connect to pull socket: Stream is closed`.

@thatch45 ideas?

**Comment by thatch45** [comment]
This does sound indicative of a larger problem, do we know if the socket is still open? Could something else in the test be closing the socket? Could we track when the socket might be getting closed?

**Comment by terminalmage** [comment]
I have re-enabled minion logging on the AWS box where I have been testing, and added some logging so we can detect when we first fail to connect to the pull socket

**Comment by terminalmage** [comment]
Note: these failures are also showing up in Ubuntu 18 now.

```
Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/support/mixins.py", line 554, in assertSaltTrueReturn
    self.assertTrue(saltret)
  File "/usr/lib/python3.6/unittest/case.py", line 682, in assertTrue
    raise self.failureException(msg)
AssertionError: False is not true

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/tmp/kitchen/testing/tests/integration/states/test_beacon.py", line 38, in test_present_absent
    self.assertSaltTrueReturn(ret)
  File "/tmp/kitchen/testing/tests/support/mixins.py", line 560, in assertSaltTrueReturn
    **(next(six.itervalues(ret)))
AssertionError: False is not True. Salt Comment:
Beacon "diskusage" is not available.	
```

**Comment by terminalmage** [comment]
OK, I was able to reproduce the same error message on a 2nd run. However, this appears to be a red herring. What seems to be happening here is that our functions to fire events first instantiate an IPC client instance, and then use that instance to fire the event. The IPC client will try until the timeout has been reached before it raises an exception (which some logging I added will catch and report the error). The problem though is that `connect_pull()` has a timeout of 1000 msec (i.e. 1 second), and the IPC client sleeps 1 second between each connection attempt. So if the first connection attempt fails, we sleep 1 second and hit the timeout.

**Comment by terminalmage** [comment]
Increasing the timeout did not work. Instead, I narrowed it down to the fact that this problem pops up before the unit tests even run, so I began running the state tests along with other single groups of integration tests. No combination of one group of integration tests followed by the state integration tests causes the beacon state integration tests to fail. So, whatever is causing this, it is not happening due to a single event. Whatever it is, it seems to be happening as part of a cumulative effect of many tests being run.
